core/input/input_event.cpp

```diff
@@ -314,14 +338,10 @@ void InputEventWithModifiers::_validate_property(PropertyInfo &p_property) const
 ///////////////////////////////////
 
 void InputEventKey::set_pressed(bool p_pressed) {
```

Comment:
> I can see the point of keeping the set_pressed(bool) API (although maybe we should consider dropping it).

We can certainly mark it as deprecated to support BC, and drop it in the next major release. Do we have an annotation or macro to mark methods as deprecated?

> However, being it only a shorthand for set_event_state(InputEventState) maybe the implementation should make that clear:; i.e., every implementation of set_pressed(bool) calling through set_event_state(InputEventState).

Do you mean to update the documentation to make that clear?

Reply:
> Do we have an annotation or macro to mark methods as deprecated?

`WARN_DEPRECATED_MSG` can be used in the setter (and getter? if it isn't too verbose).

Reply:
I don't think the `bool`-taking functions shoud be deprectared. They are very reasonable shorthands for the more verbose `InputEventState`-taking ones.

My point is that, considering `set`/`get_pressed()` as shorthands, their implementations should be always based on calling `set`/`get_state()`, with just the minimum logic necessary to map the enum to the boolean, or vice versa.

Reply:
> I don't think the `bool`-taking functions shoud be deprectared. They are very reasonable shorthands for the more verbose `InputEventState`-taking ones.
> 
> My point is that, considering `set`/`get_pressed()` as shorthands, their implementations should be always based on calling `set`/`get_state()`, with just the minimum logic necessary to map the enum to the boolean, or vice versa.

Sounds good. I believe the PR updates all implementations of `set_pressed(...)` to have it based on `set_state(...)`. `is_pressed()` is made non-virtual so it can no longer be overridden by subclasses, and its implementation is also based on `get_state()`. 
Let me know if I miss to update a location.